common-lib/src/cmr/common/util.clj

```diff
@@ -1079,14 +1080,20 @@
   "If s is a string, call read-string, otherwise returns s."
   [s]
   (if (string? s)
-    (read-string s)
+    ;; clojure.core/read-string can be used to inject runnable code.
+    ;; use clojure.edn/read-string instead.  Also since this is
+    ;; being used to convert to a number, check to see if the result
+    ;; is a number, and return it, otherwise return nil
+    (let [edn-str (edn/read-string s)]
+      (when (number? edn-str) edn-str))
     s))
 
 (defn str->num
   "If the string can be converted to a number, return that number, otherwise return nil."
   [s]
-  (if (numeric? s)
-    (safe-read-string s)
+  (if (and (string? s)
+           (numeric? s))
+      (safe-read-string s)
```

Comment:
not lined up right.  Also, where is numeric? from?  I couldn't run it.

Reply:
Fixed.  

numeric? is in the same file (common-lib/src/cmr/common/util.clj) on line 474.